workers_kv.go

```diff
@@ -82,19 +83,39 @@ func (api *API) CreateWorkersKVNamespace(ctx context.Context, req *WorkersKVName
 // ListWorkersKVNamespaces lists storage namespaces
 //
 // API reference: https://api.cloudflare.com/#workers-kv-namespace-list-namespaces
-func (api *API) ListWorkersKVNamespaces(ctx context.Context) (ListWorkersKVNamespacesResponse, error) {
-	uri := fmt.Sprintf("/accounts/%s/storage/kv/namespaces", api.AccountID)
-	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
-	if err != nil {
-		return ListWorkersKVNamespacesResponse{}, errors.Wrap(err, errMakeRequestError)
-	}
+func (api *API) ListWorkersKVNamespaces() ([]WorkersKVNamespace, error) {
+	v := url.Values{}
+	v.Set("per_page", "100")
 
-	result := ListWorkersKVNamespacesResponse{}
-	if err := json.Unmarshal(res, &result); err != nil {
-		return result, errors.Wrap(err, errUnmarshalError)
+	var namespaces []WorkersKVNamespace
+	page := 1
+
+	for {
+		v.Set("page", strconv.Itoa(page))
+		uri := fmt.Sprintf("/accounts/%s/storage/kv/namespaces?%s", api.AccountID, v.Encode())
+		res, err := api.makeRequest(http.MethodGet, uri, nil)
```

Comment:
I've replaced `api.makeRequestContext` for `api.makeRequest` per the comments on the first version of this PR. I also dropped context as an argument to this function.

Reply:
I would keep context here, we might want to eventually make all API calls context-aware (issue #324).